backend/s3/s3.go

```diff
@@ -2159,11 +2191,9 @@ func (o *Object) uploadMultipart(ctx context.Context, req *s3.PutObjectInput, si
 	)
 
 	for partNum := int64(1); !finished; partNum++ {
-		// Get a block of memory from the channel (which limits concurrency)
-		buf := <-bufs
-		if buf == nil {
-			buf = make([]byte, partSize)
-		}
+		// Get a block of memory from the pool and share from channel which limits concurrency.
+		share := <-concurrencyShares
+		buf := f.pool.Get()
```

Comment:
There is a problem here in that `partSize` may not be equal to `f.opt.ChunkSize` at this point so the memory chunks in the pool will be the wrong size.

This will happen for large files bigger than 10000*5M = 48GiB files by default.

There are several things we could do here...

1. fall back to `make([]byte, partSize)`
2. make a local `Pool` and call `Flush` on it at the end.
3. make a map of `Pool`s in `f` by partSize

What do you think? 2. is simple and would probably work OK given that there will be ~10,000 chunks used

Reply:
Nice catch. 1 and 2 would bring back initial issue this PR is trying to resolve and 3 would work, i'll implement that.